src/lib.rs

```diff
@@ -22,6 +23,16 @@ pub fn escape(s: Cow<str>) -> Cow<str> {
     }
 }
 
+/// Escape characters that may have special meaning in a shell
+/// for an `OsStr`.
+pub fn escape_os_str(s: &OsStr) -> Cow<'_, OsStr> {
+    if cfg!(unix) || env::var("MSYSTEM").is_ok() {
+        unix::escape_os_str(s)
+    } else {
+        unimplemented!("windows::escape_os_str")
```

Comment:
I don't think this can land without an implementation here.

Reply:
Yes, I think you can use [std::os::windows::ffi::OsStrExt::encode_wide](https://doc.rust-lang.org/std/os/windows/ffi/trait.OsStrExt.html) here to do the escaping then convert it back using [std::os::windows::ffi::OsStringExt::from_wide](https://doc.rust-lang.org/std/os/windows/ffi/trait.OsStringExt.html).

Reply:
@NobodyXu I've added a first draft of an implementation of `escape_os_str` for Windows. Whenever you can find the time, I'd really appreciate a review and some assistance with a test case for `test_escape_os_str_from_bytes`.